src/main/java/org/minions/devfund/reinaldo/WhackAMole.java

```diff
@@ -0,0 +1,184 @@
+     */
+    public WhackAMole(int numAttempts, int gridDimension) {
+        attemptsLeft = numAttempts;
+        this.gridDimension = gridDimension;
+        createTheMoleGrid();
+    }
+
+    /**
+     * Methods to fill the mole grid for the game whack a mole.
+     *
+     */
+    private void createTheMoleGrid() {
+        moleGrid = new char[gridDimension][gridDimension];
+        for (int i = 0; i < gridDimension; i++) {
+            for (int j = 0; j < moleGrid[i].length; j++) {
+                moleGrid[i][j] = DEFAULTCHARACTER;
+            }
+        }
+    }
+
+    /**
+     * This is the location for a wack a mole.
+     * @param posX position in x.
+     * @param posY position.
+     * @return true or false in the location from (x,y).
+     */
+    public boolean place(int posX, int posY) {
+        if (moleGrid[posX][posY] != MOLE) {
+            moleGrid[posX][posY] = MOLE;
+            molesLeft++;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * location, take a whack at that location. If that location.
+     * @param posX is the location.
+     * @param posY is the location
+     */
+    public void whack(int posX, int posY) {
+        if (moleGrid[posX][posY] == MOLE) {
+            updateForScoreMolesLeft();
+            updateForTheGrid(posX, posY);
+            return;
+        }
+        updateForTheGrid(posX, posY);
+    }
+
+    /**
+     * This methods update the molesGrid and attemptsLeft.
+     * @param posX this is the position x.
+     * @param posY this is the position y.
+     */
+    public  void updateForTheGrid(int posX, int posY) {
+        moleGrid[posX][posY] = WHACKED;
+        attemptsLeft--;
+    }
+
+    /**
+     * This method update the score and moles left.
+     */
+    public void updateForScoreMolesLeft() {
+        score++;
+        molesLeft--;
+    }
+
+    /**
+     * Methods that print the grid.
+     *
+     * @return grid string.
+     */
+    public String printGrid() {
+        StringBuilder grid = new StringBuilder();
+        for (char[] row : moleGrid) {
+            for (char value : row) {
+                addTheValueOnStringBuild(grid, value);
+            }
+            grid.append(NEWLINE);
+        }
+        return grid.toString();
+    }
+
+    /**
+     * Method that add to string build.
+     * @param grid string build.
+     * @param value char.
+     */
+    private void addTheValueOnStringBuild(final StringBuilder grid, final char value) {
+        grid.append(String.valueOf(value).concat(SPACES));
+    }
+
+    /**
+     * Methods that print the grid to user.
+     *
+     * @return string to user.
+     */
+    public String printGridToUser() {
+        StringBuilder userGrid = new StringBuilder();
+        for (char[] row : moleGrid) {
+            for (char value : row) {
+                existTheMole(userGrid, value);
+            }
+            userGrid.append(NEWLINE);
+        }
+
+        return userGrid.toString();
```

Comment:
This may be refactored like this.
```java
    public String printGridToUser() {
        StringBuilder userGrid = new StringBuilder();
        for (char[] row : moleGrid) {
            for (char value : row) {
                userGrid.append(value == MOLE ? DEFAULTCHARACTER : value).append(SPACES);
            }
            userGrid.append(NEWLINE);
        }

        return userGrid.toString();
    }
```

Reply:
thanks carlos